### PR TITLE
Fix current CI failures on push

### PR DESCRIPTION
**Pull request by t8m**

### CODE REVIEW

**Comment by t8m** [comment]
The last commit is to be dropped when merging.

**Review by hlandau — APPROVED** [review]

**Comment by hlandau** [comment]
I am skeptical about the value of `-Wmissing-braces`. `{0}` is entirely valid.

**Comment by t8m** [comment]
> I am skeptical about the value of `-Wmissing-braces`. `{0}` is entirely valid.

Yeah, I hate it as well. We could remove it from the strict warning set.

**Comment by t8m** [comment]
Instead of adding the braces I've switched-off the warning.

File: doc/designs/quic-design/record-layer.md [hunk 1]
```diff
@@ -507,7 +507,7 @@ struct ossl_record_method_st {
      * multiple records in one go and buffer them.
      */
     int (*read_record)(OSSL_RECORD_LAYER *rl, void **rechandle, int *rversion,
-                      int *type, unsigned char **data, size_t *datalen,
+                      uint8_t *type, unsigned char **data, size_t *datalen,
```
Comment (mattcaswell) [inline doc/designs/quic-design/record-layer.md:510]: If we're going to change this type I think it has ripple effects in libssl, e.g. I would also suggest changing the type of `type` and `recvd_type` in `ssl3_read_bytes` and `dtls1_read_bytes`

Comment (t8m) [reply doc/designs/quic-design/record-layer.md:510]: AFAIK these are assigned and not passed as pointer to them so type conversion should happen fine. But I can change it anyway if you prefer that.

Comment (t8m) [reply doc/designs/quic-design/record-layer.md:510]: @mattcaswell it was quite spread throughout the code but AFAIK done now, please look.

Comment (mattcaswell) [reply doc/designs/quic-design/record-layer.md:510]: > it was quite spread throughout the code but AFAIK done now, please look.

Yeah - this is what I meant by a ripple effect.

> But I can change it anyway if you prefer that.

I really dislike this type dissonance where we use different types for the same thing in different places. Looks much better now.

**Review by mattcaswell — APPROVED** [review]

**Review by hlandau — APPROVED** [review]

**Comment by hlandau** [comment]
Merged to master. Thank you.

**Pull request closed without merging**